src/components/ha-button-menu.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  customElement,
+  html,
+  TemplateResult,
+  LitElement,
+  CSSResult,
+  css,
+  query,
+} from "lit-element";
+import "@material/mwc-button";
+import "@material/mwc-menu";
+import "@material/mwc-list/mwc-list-item";
+import type { Menu } from "@material/mwc-menu";
+
+import "./ha-icon-button";
+
+@customElement("ha-button-menu")
+export class HaButtonMenu extends LitElement {
+  public icon?: string;
+
+  public text?: string;
+
+  @query("mwc-menu") private _menu?: Menu;
+
+  protected render(): TemplateResult {
+    return html`
+      ${this.icon
+        ? html`
+            <ha-icon-button
+              .icon=${this.icon}
+              @click=${this._handleClick}
+            ></ha-icon-button>
+          `
+        : this.text
```

Comment:
Let's just show a `mwc-button` when no icon, also if there is no text

Reply:
Thinks some more about this, let's make it a second slot, so the user can decide what to show as trigger.